### PR TITLE
Add deployment instructions

### PR DESCRIPTION
**Pull request by joj0s**
This PR adds a guide to deploy the application in an Ubuntu Server environment. Instructions included for:
- PostgreSQL setup
- Django project configuration and virtual environment setup
- Configuration of NginX and Gunicorn
- Redis installation
- Celery daemonization

These instructions are going to have to be updated in the near future to include reading config variables from a file instead of hardcoding them, and also possible configuration of further external applications, such as credential files used for Google Drive authentication. Also some configuration might need to change when we implement periodic datasource updates.

### CODE REVIEW

**Comment by joj0s** [comment]
Edit: I should have had master as the base branch for this one instead of the 'automatic_mappings' branch. So maybe the base branch for this PR should be changed to avoid reviewing irrelevant diffs, or maybe I should just cherry-pick the last two commits?

File: docs/deployment-ubuntu.md [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+
+```
+
+3. Next, create a psql user for our project.
+
+```
+CREATE USER ubuntu WITH PASSWORD 'password';
+```
+
+ **Important: PostgreSQL uses an authentication scheme called “peer authentication” for local connections. Basically, this means that if the user’s operating system username matches a valid Postgres username, that user can login with no further authentication. So make sure that the PSQL username matches the host OS username**:
```
[the remark this answers is not in the document]
Comment (joj0s) [reply docs/deployment-ubuntu.md:29]: Yes, this assumes that the authentication method used for PSQL is not changed.

File: docs/deployment-ubuntu.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+6. When you are finished, exit out of the PostgreSQL prompt by typing:
+
+```
+\q
+```
+
+## 3. Clone the project's code and create a virtual environment
+
+1. Make sure there is an ssh key with access to the [GitHub repo](https://github.com/EBIvariation/trait-curation). You can create an SSH key for the host machine by following [these instructions](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).
```
[the remark this answers is not in the document]
Comment (joj0s) [reply docs/deployment-ubuntu.md]: You are right, I removed this step from the guide.

File: docs/deployment-ubuntu.md [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+```
+
+2. Use the following settings
+```
+[Unit]
+Description=gunicorn daemon
+After=network.target
+
+[Service]
+User=jojos
```
[the remark this answers is not in the document]
Comment (joj0s) [reply docs/deployment-ubuntu.md]: Indeed, forgot to change that, fixed!

**Review by tcezard — APPROVED** [review]

**Pull request merged**